mlflow/store/sqlalchemy_store.py

```diff
@@ -55,15 +55,15 @@ def __init__(self, db_uri, default_artifact_root):
 
         :param db_uri: SQL connection string used by SQLAlchemy Engine to connect to the database.
                        Argument is expected to be in the format:
-                       ``db_type://<user_name>:<password>@<host>:<port>/<database_name>`
+                       ``db_type+driver://<user_name>:<password>@<host>:<port>/<database_name>`
                        Supported database types are ``mysql``, ``mssql``, ``sqlite``,
-                       and ``postgresql``.
+                       and ``postgresql``. The driver element is optional.
         :param default_artifact_root: Path/URI to location suitable for large data (such as a blob
                                       store object, DBFS path, or shared NFS file system).
         """
         super(SqlAlchemyStore, self).__init__()
         self.db_uri = db_uri
-        self.db_type = urllib.parse.urlparse(db_uri).scheme
+        self.db_type = urllib.parse.urlparse(db_uri).scheme.split('+')[0]
```

Comment:
If `driver` is specified, can we add some validation to what is acceptable in `driver` string? Something like this

```
_BAD_DRIVER_STRING_MSG = "DBAPI driver name can only include alphanumerics, underscores (_), and dashes (-)"
_VALID_DRIVER_STRING = re.compile(r"[/\w\-_]*$")
def _validate_driver(driver):
  # validates DBAPI driver name used in db_uri parameter
  if not _VALID_DRIVER_STRING.match(driver):
    raise MlflowException("Invalid DBAPI driver string: '%s'. %s" % (driver, _BAD_DRIVER_STRING_MSG), INVALID_PARAMETER_VALUE)
```

Similar examples in [mlflow.utils.validation](https://github.com/mlflow/mlflow/blob/master/mlflow/utils/validation.py)